### DISCUSSION TITLE
How would I hit GoAlert's API to determine an on-call user's email address?

### DISCUSSION THREAD

**Discussion by GraysonCAdams** [target/goalert#3839] [Q&A]
The Generic API for services seems to only allow alert creation.

Is there a way for me to reach out to GoAlert when I receive an alert to:
1. Determine who is on call
2. Get their metadata such as their email

**Comment by BenStormer** [comment]
Someone from the GoAlert team may be able to give a better suggestion, but my team recently ran into a similar situation.

It looks like you can make requests to a graphql API for goalert (`.../api/graphql`) with lots of different queries. I couldn't find documentation anywhere, but I was able to find a list of queries by searching in the Github repo with this link: https://github.com/search?q=repo%3Atarget%2Fgoalert%20const%20query%20%3D%20gql&type=code

From a quick glance, it looks like you may be able to rig your own sort of logic using the following queries:
- https://github.com/target/goalert/blob/8a7b4fae6705d8de12936ef4e4cb9be5169eb84a/web/src/app/users/UserList.tsx#L14
- https://github.com/target/goalert/blob/8a7b4fae6705d8de12936ef4e4cb9be5169eb84a/web/src/app/schedules/ScheduleList.tsx#L11
- https://github.com/target/goalert/blob/8a7b4fae6705d8de12936ef4e4cb9be5169eb84a/web/src/app/schedules/ScheduleDetails.tsx#L22
- https://github.com/target/goalert/blob/8a7b4fae6705d8de12936ef4e4cb9be5169eb84a/web/src/app/rotations/RotationList.tsx#L11

These may not actually be helpful, and I'm not sure what logic you would have to implement to reach your goals, but I bet there is some way to make something work. Hopefully this is helpful!

**Reply by smurraystrata** [reply]
You can use api/graphql/explore to click the book at the top right to be able to see all the options and requirements for graphql queries, plus test them.

There is one thing I can't figure out though - the API keys for graphql. I have a query that works in the query explorer, but I keep getting "wrong query for API key" when I try to make an api call outside the graphql explorer, despite creating the api key as admin and putting the exact query in the query box. Ideally, I could make one that allows any query, or a list of the ones I need.

Any help on creating API keys?
